martha/martha_v3.js

```diff
@@ -426,37 +427,28 @@ function overlapFields(requestedFields, serviceFields) {
     return requestedFields.filter((field) => serviceFields.includes(field)).length !== 0;
 }
 
-function sendResponse(res, requestedFields, drsResponse, fileName, bondProvider, bondSA, accessUrl) {
-    const fullResponse =
-        requestedFields.length
-            ? convertToMarthaV3Response(drsResponse, fileName, bondProvider, bondSA, accessUrl)
-            : {};
-    const partialResponse = mask(fullResponse, requestedFields.join(","));
+function buildRequestInfo(params) {
+    const {
+        url,
+        requestedFields,
+        auth,
+        userAgent,
+        ip,
+    } = params;
 
-    res.status(200).send(partialResponse);
-}
-
-// TODO: Is there some consistent way we can make this function less complex?
-//  For example: is there a way we can have utility functions that send errors and exit this outer function?
-// eslint-disable-next-line complexity
-async function marthaV3Handler(req, res) {
-    const {url, fields: requestedFields = MARTHA_V3_DEFAULT_FIELDS} = parseRequest(req);
-    const {authorization: auth, 'user-agent': userAgent} = req.headers;
-    console.log(`Received URL '${url}' from agent '${userAgent}' on IP '${req.ip}'`);
+    console.log(`Received URL '${url}' from agent '${userAgent}' on IP '${ip}'`);
 
     try {
         validateRequest(url, auth, requestedFields);
     } catch (error) {
-        logAndSendBadRequest(res, error);
-        return;
+        throw new BadRequestError(error);
```

Comment:
`validateRequest` could throw `BadRequestError`s, and then this function doesn't need to `try`/`catch`.